From 90717982978be935b9f3dd5ef36d7fa219fc2859 Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Mon, 23 Jun 2014 19:58:28 +0200
Subject: [PATCH 255/290] gpu: drm: xylon: Fixed property enable/disable

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Fixed layer update property to be disabled/enabled with the same logic
as all others properties of that type.

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
(cherry picked from commit a646bb991e521cb328b7ba9651b1c102ca2797cb)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xylon/xylon_logicvc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_logicvc.c b/drivers/gpu/drm/xylon/xylon_logicvc.c
index 4476eba..9f0632f 100644
--- a/drivers/gpu/drm/xylon/xylon_logicvc.c
+++ b/drivers/gpu/drm/xylon/xylon_logicvc.c
@@ -646,9 +646,9 @@ void xylon_cvc_ctrl(struct xylon_cvc *cvc, enum xylon_cvc_control ctrl,
 	switch (ctrl) {
 	case LOGICVC_LAYER_UPDATE:
 		if (val)
-			cvc->ctrl |= LOGICVC_CTRL_DISABLE_LAYER_UPDATE;
-		else
 			cvc->ctrl &= ~LOGICVC_CTRL_DISABLE_LAYER_UPDATE;
+		else
+			cvc->ctrl |= LOGICVC_CTRL_DISABLE_LAYER_UPDATE;
 		break;
 	case LOGICVC_PIXEL_DATA_TRIGGER_INVERT:
 		if (val)
-- 
2.0.2

